dask/dataframe/io/orc.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import absolute_import, division, print_function
+
+from ..core import DataFrame
+from ...base import tokenize
+from ...bytes.core import get_fs_token_paths
+from ...utils import import_required
+from .parquet import _get_pyarrow_dtypes, _meta_from_dtypes
+from ...compatibility import string_types
+
+__all__ = ('read_orc',)
+
+
+def _read_orc_stripe(fs, path, stripe, columns=None):
+    """Pull out specific data from specific part of ORC file"""
+    orc = import_required('pyarrow.orc', 'Please install dev version pyarrow')
+    with fs.open(path, 'rb') as f:
+        o = orc.ORCFile(f)
+        return o.read_stripe(stripe, columns).to_pandas()
+
+
+def read_orc(path, columns=None, storage_options=None):
+    """Read dataframe from ORC file(s)
+
+    Parameters
+    ----------
+    path: str or list(str)
+        Location of file(s), which can be a full URL with protocol specifier,
+        and may include glob character if a single string.
+    columns: None or list(str)
+        Columns to load. If None, loads all.
+    storage_options: None or dict
+        Further parameters to pass to the bytes backend.
+
+    Returns
+    -------
+    Dask.DataFrame (even if there is only one column)
+
+    Examples
+    --------
+    >>> d = read_orc('https://github.com/apache/orc/raw/'
+    ...              'master/examples/demo-11-zlib.orc')  # doctest: +SKIP
+    """
+    orc = import_required('pyarrow.orc', 'Please install dev version pyarrow')
+    if isinstance(path, string_types) and path.endswith('/'):
+        path = path + '*.orc'
```

Comment:
This won't work on windows (although the orc reader doesn't work on windows either currently). Please use the `os.path` api instead of string concatenation and `endswith`.

Reply:
Good point. What I actually want to do is "is this a directory", which is not automatically doable for remote backends (which happen to all use `'/'`, I think). 

Windows does handle '/'-style paths fine, though, so maybe there could be a "rationalize win paths" thing that's done for any path without a protocol on windows only. Fastparquet does exactly that internally (because it also needs to care about relative paths between files, which should be posix-style).

Reply:
On reflection, maybe we should simply not do this: if you want all the orc files in a directory, you must provide an explicit glob string. Is that OK with you, @jcrist ?

Reply:
For now that's fine, yes.